public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -56,7 +57,9 @@ const ELASTIC_META_FIELDS = [
   '_meta',
 ];
 
-export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, ElasticsearchOptions> {
+export class ElasticDatasource
+  extends DataSourceApi<ElasticsearchQuery, ElasticsearchOptions>
+  implements DataSourceWithLogsContextSupport {
```

Comment:
```suggestion
export class ElasticDatasource
  extends DataSourceWithLogs<ElasticsearchQuery, ElasticsearchOptions> {
```

Reply:
I guess one problem with this would be if we would like to support multiple capabilities e.g. if you would like to have a datasource that support both logs and some other future type.

Reply:
Just had a look at the code again and ignore the suggestion above. I think it would make more sense to introduce another interface that represents a DataSourceApi instead of relying on an abstract class everywhere.

We could have an abstract class that implements that interface for convenience but this feels like something we might need to look into within the plugins platform squad.